### PR TITLE
Grep amendments

### PR DESCRIPTION
**Pull request by mgorny-morecode**
### Description

Describe the proposed changes:

  - Remove duplicated code
  - Remove saving to file when string action
  - Changed bare strings to Input.* in params.get
  - Removed unused function

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [x] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [x] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [x] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_base64\ncat '/tmp/tmp_dvym1ub/tmp.txt' | egrep 7 \u003e /tmp/tmp_dvym1ub/results.txt\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 3,
      "matches": [
        "John,Doe,120 jefferson st.,Riverside, NJ, 08075",
        "\"John \"\"Da Man\"\"\",Repici,120 Jefferson St.,Riverside, NJ,08075",
        "Stephen,Tyler,\"7452 Terrace \"\"At the Plaza\"\" road\",SomeTown,SD, 91234"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/action_match_base64_csv.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_base64\ncat '/tmp/tmp10rc31wc/tmp.txt' | egrep hello \u003e /tmp/tmp10rc31wc/results.txt\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 2,
      "matches": [
        "hello there ",
        " hello world"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/action_match_base64_default.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_base64\ncat '/tmp/tmptw5bldsu/tmp.txt' | egrep -o hello \u003e /tmp/tmptw5bldsu/results.txt\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 2,
      "matches": [
        "hello",
        "hello"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/action_match_base64_onlyMatch.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_string\necho 'hello world \n whatup \n  hello2 world' | egrep hello \u003e /tmp/tmpt8ijjrqg/results.txt\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 2,
      "matches": [
        "hello world ",
        "  hello2 world"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/action_match_string_default.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Grep:1.0.4. Step name: match_string\necho 'hello world \n whatup \n  hello2 world' | egrep -o hello \u003e /tmp/tmppp2s7j82/results.txt\n",
    "meta": {},
    "output": {
      "found": true,
      "hits": 2,
      "matches": [
        "hello",
        "hello"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/grep:1.0.4 run < tests/action_match_string_onlyMatch.json
</summary>
</details>

<details>

```
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
Validator AcronymValidator failed!
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/icon_validator/validate.py", line 29, in validate
    v.validate(spec)
  File "/usr/local/lib/python3.7/dist-packages/icon_validator/rules/acronym_validator.py", line 90, in validate
    raise Exception(err_msg)
Exception: Acronyms found in plugin.spec.yaml that should be capitalized: ['jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq']
Acronyms found in help.md that should be capitalized: ['jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq', 'jq']
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
Output violations: Action/Trigger -> "Run": Missing ['|json_out|string|True|The output JSON|'] in help.md
Validator HelpInputOutputValidator failed!
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/icon_validator/validate.py", line 29, in validate
    v.validate(spec)
  File "/usr/local/lib/python3.7/dist-packages/icon_validator/rules/help_input_output_validator.py", line 97, in validate
    raise Exception("Help.md is not in sync with plugin.spec.yaml. Please check and rectify above violations")
Exception: Help.md is not in sync with plugin.spec.yaml. Please check and rectify above violations
[*] Executing validator DockerValidator
[*]Plugin failed validation!

----
[*] Total time elapsed: 15174.663ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] For Markdown linting, install mdl: gem install mdl

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.5
Run started:2019-12-14 22:35:13.166001

Test results:
>> Issue: [B404:blacklist] Consider possible security implications associated with subprocess module.
   Severity: Low   Confidence: High
   Location: ./icon_jq/actions/run_jq/action.py:4
   More Info: https://bandit.readthedocs.io/en/latest/blacklists/blacklist_imports.html#b404-import-subprocess
3       # Custom imports below
4       import subprocess
5       from subprocess import PIPE

--------------------------------------------------
>> Issue: [B404:blacklist] Consider possible security implications associated with PIPE module.
   Severity: Low   Confidence: High
   Location: ./icon_jq/actions/run_jq/action.py:5
   More Info: https://bandit.readthedocs.io/en/latest/blacklists/blacklist_imports.html#b404-import-subprocess
4       import subprocess
5       from subprocess import PIPE
6       import json

--------------------------------------------------
>> Issue: [B603:subprocess_without_shell_equals_true] subprocess call - check for execution of untrusted input.
   Severity: Low   Confidence: High
   Location: ./icon_jq/actions/run_jq/action.py:37
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b603_subprocess_without_shell_equals_true.html
36              self.logger.info("Command to Run: " + str(jq_cmd_array))
37              process = subprocess.Popen(jq_cmd_array, stdout=PIPE, stderr=PIPE, stdin=PIPE)
38              std_out, std_err = process.communicate(input=json.dumps(json_in).encode(), timeout=timeout)

--------------------------------------------------

Code scanned:
        Total lines of code: 140
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 3.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 3.0
Files skipped (0):
[FAIL] Fails bandit security checks
```

<summary>
make validate
</summary>
</details>

### CODE REVIEW

File: grep/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM komand/python-3-slim-plugin:2
+FROM komand/python-3-37-slim-plugin
```
Comment (jschipp-r7) [inline grep/Dockerfile]: Let's pin this to version 3
```suggestion
FROM komand/python-3-37-slim-plugin:3
```

**Comment by jschipp-r7** [comment]
Missing validation in the PR body. Run `tools/update-tools.sh` to get the latest tooling and then run  `make validate`.
